frontend/packages/client/src/hooks/useCommunity.js

```diff
@@ -37,12 +43,112 @@ export default function useCommunity() {
     }
   }, [dispatch, notifyError]);
 
-  useEffect(() => {
-    getCommunities();
-  }, [getCommunities]);
+  const createCommunity = useCallback(
+    async (injectedProvider, communityData) => {
+      dispatch({ type: "PROCESSING" });
+      const url = `${process.env.REACT_APP_BACK_END_SERVER_API}/communities`;
+      try {
+        const timestamp = Date.now().toString();
+        const hexTime = Buffer.from(timestamp).toString("hex");
+        const _compositeSignatures = await injectedProvider
+          .currentUser()
+          .signUserMessage(hexTime);
+
+        const compositeSignatures = getCompositeSigs(_compositeSignatures);
+
+        if (!compositeSignatures) {
+          const statusText = "No valid user signature found.";
+          // modal error will open
+          notifyError(
+            {
+              status: "Something went wrong with creating the community.",
+              statusText,
+            },
+            url
+          );
+          dispatch({
+            type: "ERROR",
+            payload: { errorData: statusText },
+          });
+          return;
+        }
+
+        const {
+          communityName: name,
+          communityDescription: body,
+          category,
+          communityTerms: termsAndConditionsUrl,
+          listAddrAdmins,
+          listAddrAuthors,
+          creatorAddr,
+          slug,
+          proposalThreshold,
+          discordUrl,
+          githubUrl,
+          instagramUrl,
+          twitterUrl,
+          websiteUrl,
+          logo,
+          contractAddress,
+          contractName,
+          storagePath,
+          onlyAuthorsToSubmitProposals,
```

Comment:
@jacksonConrad I have added these fields to manage the contract information. Let me know if field names need to be change to match the payload validation

Reply:
@germanurrus @jacksonConrad do we have a ticket for the BE work needed here?

Reply:
I think it's `cas-69`